appveyor.yml

```diff
@@ -0,0 +1,78 @@
+image:
+- Visual Studio 2017
+
+environment:
+  matrix:
+    # For Python versions available on Appveyor, see
+    # http://www.appveyor.com/docs/installed-software#python
+    - PYTHON: "C:\\Python27"
+      WIN_ARCH: "x86"
+      SETUP_PY: "setup.py"
+    - PYTHON: "C:\\Python27-x64"
+      WIN_ARCH: "amd64"
+      SETUP_PY: "setup.py"
+    - PYTHON: "C:\\Python33"
+      WIN_ARCH: "x86"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python33-x64"
+      WIN_ARCH: "amd64"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python34"
+      WIN_ARCH: "x86"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python34-x64"
+      WIN_ARCH: "amd64"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python35"
+      WIN_ARCH: "x86"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python35-x64"
+      WIN_ARCH: "amd64"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python36"
+      WIN_ARCH: "x86"
+      SETUP_PY: "setup3.py"
+    - PYTHON: "C:\\Python36-x64"
+      WIN_ARCH: "amd64"
+      SETUP_PY: "setup3.py"
+
+install:
+  # Prepare environment
+  - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
+
+  # Check that we have the expected version and architecture for Python
+  - "%PYTHON%\\python.exe --version"
+
+  # Install setuptools/wheel so that we can e.g. use bdist_wheel
+  - "pip install setuptools wheel"
+
+build_script:
+  - "CALL \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" %WIN_ARCH%"
+  - "%PYTHON%\\python.exe %SETUP_PY% build"
+
+test_script:
+  # Put your test command here.
+  # If you don't need to build C extensions on 64-bit Python 3.3 or 3.4,
+  # you can remove "build.cmd" from the front of the command, as it's
+  # only needed to support those cases.
+  # Note that you must use the environment variable %PYTHON% to refer to
+  # the interpreter you're using - Appveyor does not do anything special
+  # to put the Python version you want to use on PATH.
+  - "%PYTHON%\\python.exe pywin32_testall.py"
+
+after_test:
+  # This step builds your wheels.
+  # Again, you only need build.cmd if you're building C extensions for
+  # 64-bit Python 3.3/3.4. And you need to use %PYTHON% to get the correct
+  # interpreter
+  - "CALL \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" %WIN_ARCH%"
+  - "%PYTHON%\\python.exe %SETUP_PY% bdist_wheel"
```

Comment:
Appveyor is capable of showing you a list of tests separately from the console output:
```yaml
after_test:
  - ps: |
      $wc = New-Object 'System.Net.WebClient'
      $wc.UploadFile("https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path .\junit-test-results.xml))  # make sure your test runner puts junit-compatible test results into this file
```

Reply:
Sounds interesting! Personally I don't have much experiences with Appveyor, but do you know how it comes that I'm getting `MemoryError`s? In my point of view I'm doing the same steps, like in my local setup, but at Appveyor I'm getting these errors.

Reply:
Dunno, you probably hit some memory limit there. I don't have insights on this, but you should be able to report this to Appveyor crew.

Reply:
@thopiekar BTW, I think you don't need `setup3.py`: it's been possible to write cross-compatible code for ages. Just fix a few bits of syntax (print etc?) and imports (try/except ImportError + drop distutils in favour of modern setuptools) in `setup.py`

Reply:
I know. My intention was to get this CI functionality into pywin32 first and then begin to contribute with cleanups and updating some PRs I already made.
I think it is also very useful for other people to know early whether their changes work on all supported python versions or not.

Reply:
you can still enable Appveyor tests, but don't make failing CI prevent PR from merges until this is fixed

Reply:
https://www.appveyor.com/docs/build-environment/#build-vm-configurations it seems they provide you with enough memory

Reply:
Strange, so either a memory leak on PyWin32's side or something else I missed or don't see on my own.

Reply:
memory leak looks happening in `find_platform_sdk_dir` function, maybe it's related to the fact you do `exec`, i don't know. I'd start with just logging memory usage at various points + removing `exec()` in favor of native script execution. btw, I'd disable creation of `*.pyc` files for test purposes as well. (via env var)